src/astro/databases/base.py

```diff
@@ -0,0 +1,259 @@
+        """
+        Return a Sqlalchemy connection object for the given database.
+        """
+        return self.sqlalchemy_engine.connect()
+
+    @property
+    def sqlalchemy_engine(self) -> sqlalchemy.engine.base.Engine:
+        """
+        Return Sqlalchemy engine.
+        """
+        return self.hook.get_sqlalchemy_engine()
+
+    def run_sql(self, sql_statement: str, parameters: Optional[dict] = None):
+        """
+        Return the results to running a SQL statement.
+
+        Whenever possible, this method should be implemented using Airflow Hooks,
+        since this will simplify the integration with Async operators.
+
+        :param sql_statement: Contains SQL query to be run against database
+        :param parameters: Optional parameters to be used to render the query
+        """
+        return self.hook.run(sql_statement, parameters)
+
+    # ---------------------------------------------------------
+    # Table metadata
+    # ---------------------------------------------------------
+    def get_table_qualified_name(self, table: Table) -> str:
+        """
+        Return table qualified name. This is Database-specific.
+        For instance, in Sqlite this is the table name. In Snowflake, however, it is the database, schema and table
+
+        :param table: The table we want to retrieve the qualified name for.
+        """
+        # Initially this method belonged to the Table class.
+        # However, in order to have an agnostic table class implementation,
+        # we are keeping all methods which vary depending on the database within the Database class.
+        raise NotImplementedError
+
+    # ---------------------------------------------------------
+    # Table creation & deletion methods
+    # ---------------------------------------------------------
+    def create_table(self, table: Table) -> None:
+        """
+        Create a SQL table. For this method to work, the table instance must contain columns.
+
+        :param table: The table to be created.
+        """
+        metadata = table.sqlalchemy_metadata
+        sqlalchemy_table = sqlalchemy.Table(table.name, metadata, *table.columns)
+        metadata.create_all(self.sqlalchemy_engine, tables=[sqlalchemy_table])
+
+    def create_table_from_select_statement(
+        self,
+        statement: str,
+        target_table: Table,
+        parameters: Optional[dict] = None,
+    ) -> None:
+        """
+        Export the result rows of a query statement into another table.
+
+        :param statement: SQL query statement
+        :param target_table: Destination table where results will be recorded.
+        :param parameters: (Optional) parameters to be used to render the SQL query
+        """
+        statement = self._create_table_statement.format(
+            self.get_table_qualified_name(target_table), statement
+        )
+        self.hook.run(statement, parameters)
+
+    def drop_table(self, table: Table) -> None:
```

Comment:
Would it be better to move this method to Table Class?

Reply:
This method is currently customizable per database. As an example, BQ does not support `IF EXISTS` in the drop table statement. Since we are aiming to have an agnostic Table class, how do you suggest we move the `drop_table` functionality there?